src/main/java/de/retest/recheck/execution/RecheckAdapters.java

```diff
@@ -4,18 +4,20 @@
 import java.util.stream.StreamSupport;
 
 import de.retest.recheck.RecheckAdapter;
+import de.retest.recheck.RecheckOptions;
 
 public class RecheckAdapters {
 
 	private static final ServiceLoader<RecheckAdapter> adapters = ServiceLoader.load( RecheckAdapter.class );
 
 	private RecheckAdapters() {}
 
-	public static RecheckAdapter findAdapterFor( final Object toVerify ) {
+	public static RecheckAdapter findAdapterFor( final Object toVerify, final RecheckOptions options ) {
 		return StreamSupport.stream( adapters.spliterator(), false ) //
 				.filter( adapter -> adapter.canCheck( toVerify ) ) //
 				.findAny() //
-				.orElseThrow( () -> createHelpfulExceptionForMissingAdapter( toVerify.getClass().getName() ) );
+				.orElseThrow( () -> createHelpfulExceptionForMissingAdapter( toVerify.getClass().getName() ) )
+				.initialize( options );
```

Comment:
Be advised, that a `ServiceLoader` caches the classes. So you would call `initialize` multiple times&mdash;with its name suggest only be called once.

Reply:
Yes, but I don't see that as a problem. Do you?

Reply:
I just wanted to emphasize that we should discuss the lifetime of a `RecheckAdapter` as with this proposal, it only lives during a single `check`.

Reply:
I see you point, but for the Adapters we currently have, this is no problem, as they are lightweight. Anyways, the situation wouldn't be any different in a big refactoring, if this was a clean factory. It shouldn't be a problem now.